From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: NeKit <nekit1000@gmail.com>
Date: Sun, 5 Jul 2020 13:56:04 +0200
Subject: [PATCH] (halium) use 127.0.1.1 as fallback DNS server

Run by NetworkManager system in Ubuntu Touch. Needed to fix some
Android-side daemons requiring internet access like vendor AGPS.

Change-Id: I3c8dd8cb4d64d7632a853b12ff8c2f066534fcad
---
 libc/dns/resolv/res_init.c | 4 ++++
 1 file changed, 4 insertions(+)

diff --git a/libc/dns/resolv/res_init.c b/libc/dns/resolv/res_init.c
index 2fb2a3d4c..8b013fb2e 100644
--- a/libc/dns/resolv/res_init.c
+++ b/libc/dns/resolv/res_init.c
@@ -200,8 +200,12 @@ __res_vinit(res_state statp, int preinit) {
 	}
 
 	memset(u, 0, sizeof(u));
+// Halium - dnsmasq is listening at 127.0.1.1
+#define USELOOPBACK
 #ifdef USELOOPBACK
 	u[nserv].sin.sin_addr = inet_makeaddr(IN_LOOPBACKNET, 1);
+	// Ubuntu Touch specific - provided by NetworkManager
+	u[nserv].sin.sin_addr.s_addr = inet_addr("127.0.1.1");
 #else
 	u[nserv].sin.sin_addr.s_addr = INADDR_ANY;
 #endif
-- 
2.25.1

